### PR TITLE
[Search v1] Implement Bottom Tab Bar Changes

### PR DESCRIPTION
**Pull request by grgia**
### Details

https://github.com/Expensify/App/assets/38015950/c2f13338-5b94-4e01-9a79-0257c608eb89

https://github.com/Expensify/App/assets/38015950/877b7e3c-8026-449d-971c-43685b0a301f

### Fixed Issues

$ https://github.com/Expensify/App/issues/39886

### Tests

- [ ] Verify that no errors appear in the JS console
- [ ] Verify that the order of icons is Inbox, Search, Account Settings, FAB
- [ ] Click on Inbox, Verify LHN opens and icon turns green
- [ ] Hover over Inbox, verify tooltip reads "Inbox"
- [ ] Click on Search, Verify Search Route opens and icon turns green
- [ ] Hover over Search, verify tooltip reads "Search"
- [ ] Click on Profile, Verify Profile Page opens
- [ ] Click on FAB, verify quick action menu opens

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
@grgia we might either wanna hide this button or hold on merging this PR until the rest of the functionality is ready.

**Comment by grgia** [comment]
@luacmartins We could just hold it since it also moves the global create button

**Comment by grgia** [comment]
@WojtekBoman would you mind taking a look at the routes here?

**Comment by shawnborton** [comment]
Let's run a test build! 

At first glance, it seems like the size of the bottom tabs is off? I was thinking they would each be equal widths:
CleanShot 2024-04-25 at 15 25 22@2x

**Review by luacmartins — APPROVED** [review]
LGTM. Seems like @shawnborton has some comments though

**Review by luacmartins — CHANGES_REQUESTED** [review]
Ahh @grgia one thing that I just remembered, with this PR we can also remove the navigation from the troubleshoot page - https://github.com/Expensify/App/blob/8b71b1b571bea3475dceaa6c33bd372154c9ee73/src/components/TestToolMenu.tsx#L96-L105

**Comment by grgia** [comment]
@shawnborton what size/padding should the FAB have? I can fix the flex row sizing but the button is bigger than that mock

**Comment by grgia** [comment]
Done @luacmartins 

@shawnborton I fixed the even spacing and triggered a new test build for ya

**Comment by shawnborton** [comment]
Dope, thanks Georgia!

**Comment by shawnborton** [comment]
Styles look good to me! In theory, we could make tapping the entire tab area that surrounds the FAB tappable, but I don't feel too strongly since the FAB itself is quite big.

**Comment by dannymcclain** [comment]
Looking good to me too! I even have a (phantom 😩) RBR on my inbox and it looks dope.

CleanShot 2024-04-25 at 15 36 45@2x

>we could make tapping the entire tab area that surrounds the FAB tappable, but I don't feel too strongly since the FAB itself is quite big.

Agree but also don't feel strongly at all. I'd be totally fine leaving it how it is TBH.

**Comment by dubielzyk-expensify** [comment]
This is looking great! 😄

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
@fedirjh please review this one when you're online. Let's get this one approved and ready to merge for the photo finish.

**Review by fedirjh — APPROVED** [review]

**Review by dannymcclain — APPROVED** [review]
Looks like all the designers have chimed in so I'm going to go ahead and approve from a design perspective! 🚀

**Review by marcochavezf — APPROVED** [review]

**Comment by luacmartins** [comment]
hmm tests were passing when I merged it.

File: src/languages/es.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ export default {
         magicCode: 'Código mágico',
         twoFactorCode: 'Autenticación de dos factores',
         workspaces: 'Espacios de trabajo',
-        chats: 'Chats',
+        inbox: 'Bandeja de entrada',
```
Comment (s77rt) [inline src/languages/es.ts:141]: This text overflows https://github.com/Expensify/App/pull/42772#issuecomment-2156032649

Comment (luacmartins) [reply src/languages/es.ts:141]: @s77rt do we have a fix for that yet?

**Pull request merged**